plugins/by-name/clangd-extensions/default.nix

```diff
@@ -76,18 +76,34 @@ lib.nixvim.plugins.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings = lib.nixvim.mkWarnings "plugins.clangd-extensions" {
-      when = !config.plugins.lsp.enable;
+      when =
+        !(
+          (config.plugins.lsp.servers.clangd.enable or false) || (config.lsp.servers.clangd.enable or false)
+        );
 
       message = ''
-        You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
```

Comment:
I believe the reason for the old wording was that `plugins.lsp.servers.clangd` gets enabled by default, but has no effect until the user manually enables `plugins.lsp`?

Therefore the new wording should suggest enabling _either_ `plugins.lsp` or `lsp.servers.clangd`.

Maybe the warning should also note that the user probably also wants `plugins.lspconfig` if they're using the latter option?

Reply:
You're right, I've changed back. But regarding lspconfig I think we should not default to it as per the discussion in #3261.